os/kernel/binary_manager/binary_manager.c

```diff
@@ -0,0 +1,347 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <signal.h>
+#include <mqueue.h>
+#include <errno.h>
+#include <debug.h>
+#include <fcntl.h>
+#if !defined(CONFIG_DISABLE_PTHREAD)
+#include <pthread.h>
+#endif
+#ifdef CONFIG_BOARD_ASSERT_AUTORESET
+#include <sys/boardctl.h>
+#include <stdlib.h>
+#endif
+#include <tinyara/sched.h>
+#include <tinyara/binary_manager.h>
+
+
+#include "group/group.h"
+#include "binary_manager.h"
+/****************************************************************************
+ * Private Definitions
+ ****************************************************************************/
+#define bmvdbg dbg
+#define bmdbg dbg
+
+#define BINMGR_DEVNAME_FMT "/dev/mtdblock%d"
+#define BINMGR_DEVNAME_LEN 16
+
+struct binary_header_s {
+	uint16_t header_size;
+	uint16_t bin_type;
+	uint32_t bin_size;
+	char bin_name[BIN_NAME_MAX];
+	char bin_ver[BIN_VER_MAX];
+	char kernel_ver[KERNEL_VER_MAX];
+	uint32_t jump_addr;
```

Comment:
Could you please add the comments for better understanding of the variables and structure use?